### ISSUE TITLE
scan: include_simulator_logs silently fails always in recent iOS

### ISSUE THREAD

**Issue by lilyball** [fastlane/fastlane#16800]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [X] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
When passing `include_simulator_logs: true`, the log collection silently fails 100% of the time against a recent simulator. I assume it worked against iOS 12 simulators (but have not tested) because otherwise I don't know how this would have been missed, but against an iOS 13 simulator (I've tested iOS 13.4 and iOS 13.5) it produces an empty logarchive folder. Running the command by hand (without suppressing stderr) produces

```
log: failed to collect LiveData: No such file or directory (2)
Archive successfully written to /path/to/system_logs-iPhone 8_iOS_13.5.logarchive
```

The culprit here is the `--standalone` flag. This flag was introduced in https://github.com/fastlane/fastlane/pull/15440 as a misguided fix for https://github.com/fastlane/fastlane/issues/15205. The real problem with #15205 was attempting to collect logs from a shutdown device. This is what the `--standalone` flag is for, except `log collect` doesn't work in a standalone fashion, not even when the device is booted.

The correct fix here is for scan to ensure the device is booted prior to running the tests (assuming it wants to collect logs), then collect logs without `--standalone`, then shut down the device (assuming it booted it in the first place). If the device is already booted when xcode is told to run its tests, xcode will leave the device in the booted state after the tests are done (as opposed to shutting it down, which is what xcode does if it had to boot the device in the first place). By keeping it booted, it can collect the logs without `--standalone`, which should work, and then it can shut it down.

Ideally it would do a `build-for-testing` first, then boot the sim, then a `test`, then shut down the sim. This way the sim isn't running prematurely. It should also ensure that it catches any exceptions thrown during the test process (e.g. because it was signalled) in order to shut down the simulator (if it had booted it).

**Comment by lilyball** [comment]
No thanks, my `fastlane env` contains too much sensitive data to comb through, and it doesn't matter anyway for this issue.

**Comment by lyndsey-ferguson** [comment]
I'm looking into this

**Comment by lilyball** [comment]
This was fixed by #16833.
